[16.0][MIG] sale_mrp_link: Migration to 16.0 (from 12.0)

legalsylvain opened the pull request.

achaussende committed:
[ADD] sale_mrp_link module

oca-travis committed:
[UPD] Update sale_mrp_link.pot

OCA-git-bot committed:
[UPD] README.rst

tafaRU committed:
[FIX] Also separate test for sale_delivery_split_date as incompatible with sale_mrp_link

OCA-git-bot committed:
[ADD] icon.png

Commit:
[MIG] Migration of sale_mrp_link to 12.0

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update sale_mrp_link.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/

Commit:
[12.0][IMP] sale_mrp_link: One2many field on sale.order

As mrp.production already has a Many2one field pointing to sale.order it makes sense to have a One2many on the "other side".
This adds multiple benefits:
- Easier to open linked production with the smart button action
- Simpler production count computing (also now can store the result)
- Dependent modules can use the One2many field for other purposes

oca-travis committed:
[UPD] Update sale_mrp_link.pot

OCA-git-bot committed:
sale_mrp_link 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/

ypapouin committed:
Added translation using Weblate (French)

ypapouin committed:
Translated using Weblate (French)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/fr/

jvargass committed:
Added translation using Weblate (Spanish)

jvargass committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/es/

danimv5 committed:
Added translation using Weblate (Catalan)

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/ca/

yavuzavci committed:
Added translation using Weblate (Turkish)

yavuzavci committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/tr/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_mrp_link
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_mrp_link/

legalsylvain committed:
[MIG] sale_mrp_link: Migration to 16.0 (from 12.0)

legalsylvain committed:
[IMP] sale_mrp_link: pre-commit stuff

yvaucher commented:
/ocabot migration sale_mrp_link

ypapouin commented on sale_mrp_link/models/mrp_production.py:
```diff
@@ -0,0 +1,34 @@
+                )
+                # If so, use the 'sale_order_id' from the parent production
+                values["sale_order_id"] = production_id.sale_order_id.id
+
```
`sale_order_id` should not be overridden if already set (by third-party module)

Maybe add a check before:
`if "origin" in values and not values.get("sale_order_id")`:

Or browse the `sale_id` first and keep existing code:
```
if "origin" in values:
    # Checking first if this comes from a 'sale.order'
    sale_id = self.env["sale.order"].browse(values.get("sale_order_id"))
    if not sale_id:
        sale_id = self.env["sale.order"].search(
            [("name", "=", values["origin"])], limit=1
        )
```

rousseldenis commented:
@legalsylvain What's the status of this ?

legalsylvain commented:
Not sure to know how to finish this work. I didn't developped the module at the beginning, and I'm not sure to know how to address @ypapouin remark, as it is about code I didn't written.

feel free to finish !

ypapouin commented:
@legalsylvain , just replacing `if "origin" in values:` with `if "origin" in values and not values.get("sale_order_id")` should be enough.

legalsylvain committed:
[FIX] sale_mrp_link : avoid to override sale_order_id, if defined by third-party module

legalsylvain committed:
[REF] sale_mrp_link: update documentation

legalsylvain committed:
[IMP] sale_mrp_link : use new api_create_multi decorator

legalsylvain commented:
Hi @ypapouin  and @rousseldenis.

Sorry for the late. PR finished and rebased.
just two new commits to review : 

- https://github.com/OCA/sale-workflow/pull/3339/commits/90d5a238846657861521bee93d4a4a1cc62e288d : apply @ypapouin fix.
- https://github.com/OCA/sale-workflow/pull/3339/commits/f8f81e1b54ffa24ccd00c5692259f19b5be35b7e : use new api_create_multi decorator.

Could you finish your review ? 

Updated : tests are failing. no need to review for the time being.

thanks !

legalsylvain commented:
In fact, the feature is now native. closing.

https://github.com/odoo/odoo/pull/43366
https://github.com/odoo/odoo/commit/e1b3a17c3bda2b2d585d930386244e3f2b0bed31

you can review This OpenUpgrade PR : https://github.com/OCA/OpenUpgrade/pull/5562
